src/main/java/net/snowflake/client/jdbc/SnowflakeSQLLoggedException.java

```diff
@@ -69,100 +67,119 @@ private void sendOutOfBandTelemetryMessage(JSONObject value, SnowflakeSQLLoggedE
    * @param value ObjectNode containing information specific to the exception constructor that
    *     should be included in the telemetry log, such as SQLState or reason for the error
    * @param ex The exception being thrown
+   * @param ibInstance Telemetry instance through which telemetry log will be sent
    * @return true if in-band telemetry log sent successfully or false if it did not
    */
-  private Future<Boolean> sendInBandTelemetryMessage(
-      ObjectNode value, SnowflakeSQLLoggedException ex) {
+  private static Future<Boolean> sendInBandTelemetryMessage(
+      ObjectNode value, SQLException ex, Telemetry ibInstance) {
     StringWriter sw = new StringWriter();
     PrintWriter pw = new PrintWriter(sw);
     ex.printStackTrace(pw);
-    String stackTrace = sw.toString();
+    String stackTrace = maskStacktrace(sw.toString());
     value.put("Stacktrace", stackTrace);
     value.put("Exception", ex.getClass().getSimpleName());
     ibInstance.addLogToBatch(TelemetryUtil.buildJobData(value));
     return ibInstance.sendBatchAsync();
   }
 
+  /**
+   * Helper function to remove sensitive data (error message, reason) from the stacktrace.
+   *
+   * @param stackTrace original stacktrace
+   * @return
+   */
+  static String maskStacktrace(String stackTrace) {
```

Comment:
NEW CHANGE: added function to remove sensitive info from stack trace. This always appears as a message immediately after the exception type in the first line. If that message appears, it's removed.

Reply:
Maybe we can move this to SecretDetector class?

Reply:
We could, although I don't anticipate it being used elsewhere in the driver. It is pretty much a telemetry-specific function for the stacktrace in the error message. In all other cases, we'd want to see the whole stacktrace, I think.